ivadomed/transforms.py

```diff
@@ -42,148 +142,86 @@ def __call__(self, sample):
         self.wspace = wspace
-        self.interpolation = interpolation
-        self.labeled = labeled
+        self.dspace = dspace
+        self.interpolation_order = interpolation_order
 
-    @staticmethod
-    def do_resample(list_data, new_shape, interpolation_mode):
-        list_data_out = []
-        for i, data in enumerate(list_data):
-            resampled_data = data.resize(new_shape,
-                                         resample=interpolation_mode)
-            list_data_out.append(resampled_data)
-        return list_data_out
-
-    def undo_transform(self, sample):
-        rdict = {}
-
-        # Get original data shape
-        hshape, wshape = sample['input_metadata']['data_shape']
+    @multichannel_capable
+    @two_dim_compatible
+    def undo_transform(self, sample, metadata):
+        assert "resample" in metadata
 
-        # Input data
-        rdict['input'] = self.do_resample(list_data=sample["input"],
-                                          new_shape=(wshape, hshape),
-                                          interpolation_mode=self.interpolation)
+        # Get params
+        params_do = metadata["resample"]
+        params_undo = [1 / x for x in params_do]
 
-        # Prediction data
-        # Note: We use here self.interpolation instead of forcing Image.NEAREST
-        #       in order to ensure soft output
-        rdict['gt'] = self.do_resample(list_data=sample["gt"],
-                                       new_shape=(wshape, hshape),
-                                       interpolation_mode=self.interpolation)
+        # Undo resampling
+        data_out = zoom(sample,
+                        zoom=params_undo,
+                        order=self.interpolation_order)
 
-        # Update
-        sample.update(rdict)
-        return sample
+        # Data type
+        data_out = data_out.astype(sample.dtype)
 
-    def __call__(self, sample):
-        rdict = {}
+        return data_out, metadata
 
-        # Get new data shape
-        # Based on the assumption that the metadata of every modality are equal.
+    @multichannel_capable
+    @two_dim_compatible
+    def __call__(self, sample, metadata):
+        # Get params
         # Voxel dimension in mm
-        input_metadata = sample['input_metadata']
-        hzoom, wzoom = input_metadata[0]["zooms"]
-        hshape, wshape = input_metadata[0]["data_shape"]
-        hfactor = hzoom / self.hspace
-        wfactor = wzoom / self.wspace
-        hshape_new = int(round(hshape * hfactor))
-        wshape_new = int(round(wshape * wfactor))
-
-        # Input data
-        rdict['input'] = self.do_resample(list_data=sample["input"],
-                                          new_shape=(wshape_new, hshape_new),
-                                          interpolation_mode=self.interpolation)
+        zooms = metadata["zooms"]
+        if len(zooms) == 2:
+            zooms += tuple([1.0])
+        hfactor = zooms[0] / self.hspace
```

Comment:
This branch is not merged with master, so I'm getting the wrong zooms for sagittal images (so images don't respect de hwd format). The same problem applies to CenterCrop. Let me know if you need help to merge with master :) .

Reply:
should be better now :)